ambry-replication/src/test/java/com.github.ambry.replication/ReplicationTest.java

```diff
@@ -611,16 +612,19 @@ private void addDeleteMessagesToReplicasOfPartition(PartitionId partitionId, Sto
       throws MessageFormatException, IOException {
     int blobSize = TestUtils.RANDOM.nextInt(500) + 501;
     int userMetadataSize = TestUtils.RANDOM.nextInt(blobSize / 2);
+    int encryptionKeySize = TestUtils.RANDOM.nextInt(blobSize / 4);
     byte[] blob = new byte[blobSize];
     byte[] usermetadata = new byte[userMetadataSize];
+    byte[] encryptionKey = new byte[encryptionKeySize];
     TestUtils.RANDOM.nextBytes(blob);
     TestUtils.RANDOM.nextBytes(usermetadata);
     BlobProperties blobProperties = new BlobProperties(blobSize, "test", accountId, containerId);
 
-    MessageFormatInputStream stream = new PutMessageFormatInputStream(id, blobProperties, ByteBuffer.wrap(usermetadata),
-        new ByteBufferInputStream(ByteBuffer.wrap(blob)), blobSize);
+    MessageFormatInputStream stream =
+        new PutMessageFormatInputStream(id, ByteBuffer.wrap(encryptionKey), blobProperties,
```

Comment:
We have not typically done these kind of changes in the replication tests for message format version changes. It is a bit difficult to plug that in since this class is not in the message format package. Plus, I don't know if this falls under the purview of "unit tests". These should have mocked the write streams to being with, really.

Reply:
That said - I have manually verified all changes with HEADER version set to the new one at a global level.